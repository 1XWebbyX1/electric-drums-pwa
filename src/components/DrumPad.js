import React from 'react'
import $ from 'jquery';


class DrumPad extends React.Component {
  constructor(props){
    super(props);
    this.state = {
      id: ''
    }
    this.handleKeyPress = this.handleKeyPress.bind(this);
    this.handleClick = this.handleClick.bind(this);
  }

  //adding event listener _______________________________________
  componentDidMount() {
    document.addEventListener('keydown', this.handleKeyPress);
    $('.display-text').text('Electric Drums')
  }
  //-------------------------------------------------------------

  //removing event listener________________________________________
  componentWillUnmount() {
    document.removeEventListener('keydown', this.handleKeyPress);
  }
  //-----------------------------------------------------------------

  handleKeyPress(e) {
    if (e.keyCode === this.props.keyCode) {
     this.handleClick();
    }
  }

  handleClick() {
     //playing audio for a key
     const audio = document.getElementById(this.props.id);
      audio.currentTime = 0;
      audio.play();
     //----------------------

      //adding animation for current pressed key
      var str = 'div-' + this.props.id;
      $('#' + str).addClass('float');
      setTimeout(function() {
        $('#' + str).removeClass('float');
      }, 1000);
      //--------------------------------------


      //animations for display text
      $('.display-text').animate(
      { opacity: 0.5 },
      100,
      function() {
       $('.display-text').animate({ opacity: 1}, 100);
    }
  );
       $('.display-text').text(this.props.display);
       //--------------------------------
  }


  render() {
    return(
      <div id={'div-' + this.props.id} className="drum-pad" onClick={this.handleClick}>
        <h3 id={'h3-' + this.props.id}>{this.props.text}</h3>
        <audio id={this.props.id} className="clip" src={this.props.src}></audio>
      </div>
    )
  }
}

export default DrumPad;
