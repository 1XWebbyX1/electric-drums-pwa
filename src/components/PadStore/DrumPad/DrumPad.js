import React from 'react'
import $ from 'jquery'

class DrumPad extends React.Component {
  constructor(props){
    super(props);
    this.handleKeyPress = this.handleKeyPress.bind(this);
    this.handleClick = this.handleClick.bind(this);
    this.pad = React.createRef();
    this.audio = React.createRef();
  }

  //adding event listener _______________________________________
  componentDidMount() {
    document.addEventListener('keydown', this.handleKeyPress);
    $('.display-text').text('Electric drums');
  }

  //removing event listener________________________________________
  componentWillUnmount() {
    document.removeEventListener('keydown', this.handleKeyPress);
  }

  handleKeyPress(e) {
    if (e.keyCode === this.props.keyCode) {
     this.handleClick();
    }
  }

  handleClick() {
     //playing audio for a key
     const audio = this.audio.current;
     audio.currentTime = 0;
     audio.play();
     //----------------------

      //adding animation for current pressed key
      const node = this.pad.current;
      node.classList.add('float');
      setTimeout(function() {
        node.classList.remove('float');
      }, 1000);
      //--------------------------------------

      $('.display-text').animate(
        { opacity: 0.5 },
        100,
        function() {
         $('.display-text').animate({ opacity: 1}, 100);
       });

      //update text content
      $('.display-text').text(this.props.display);

  }


  render() {
    return(
      <div id={'div-' + this.props.id} className="drum-pad" onClick={this.handleClick} onKeyDown={this.handleKeyPress} ref={this.pad}>
        <h3 id={'h3-' + this.props.id}>{this.props.text}</h3>
        <audio id={this.props.id} className="clip" src={this.props.src} ref={this.audio}></audio>
      </div>
    )
  }
}

export default DrumPad;
