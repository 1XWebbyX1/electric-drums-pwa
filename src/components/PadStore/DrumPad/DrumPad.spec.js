// External Depedencies
import React from 'react';
import { shallow, mount } from 'enzyme';
// Our Dependencies
import { expect } from '../../utils/chai';
import sinon from 'sinon'
import renderer from "react-test-renderer";

// Our Component
import DrumPad from './DrumPad';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });


describe('DrumPad', () => {

  it("renders correctly", () => {
    const output = renderer.create(
        <DrumPad id="Q" text="Q" src="https://s3.amazonaws.com/freecodecamp/drums/RP4_KICK_1.mp3" keyCode={81} display="mockText" />
  ).toJSON();

    expect(output).to.matchSnapshot();
  });

  describe('when a key is pressed', () => {
    var spy;
    var output;

    beforeEach(() => {
     spy = sinon.spy(DrumPad.prototype, 'handleClick');
     output = mount(
        <DrumPad id="Q" text="Q" src="https://s3.amazonaws.com/freecodecamp/drums/RP4_KICK_1.mp3" keyCode={81} display="mockText" />
      );
    })

    afterEach(() => {
      spy.restore();
    });

    it('should call handleClick when props key equals pressed key', () => {
      output.simulate('keydown', {
        keyCode: 81
      });
      expect(spy.calledOnce).to.be.true;
    });

    it('should not call handleClick when props key not equals pressed key', () => {
      output.simulate('keydown', {
        keyCode: 69
      });
      expect(spy.notCalled).to.be.true;
    });
  })


})
