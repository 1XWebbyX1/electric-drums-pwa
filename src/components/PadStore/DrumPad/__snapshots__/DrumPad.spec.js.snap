// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrumPad renders correctly 1`] = `
<div
  className="drum-pad"
  id="div-Q"
  onClick={[Function]}
  onKeyDown={[Function]}
>
  <h3
    id="h3-Q"
  >
    Q
  </h3>
  <audio
    className="clip"
    id="Q"
    src="https://s3.amazonaws.com/freecodecamp/drums/RP4_KICK_1.mp3"
  />
</div>
`;
